### DISCUSSION TITLE
Problem using "relationship" in Lists and Views

### DISCUSSION THREAD

**Discussion by infiniteroles** [Laravel-Backpack/community-forum#1124] [Q&A (Help)] [answered]
Hello, I am trying to display a column in the list where the wallet addresses associated with a user appear.

In the User model, I have:

public function wallets()
```
{
    return $this->hasMany(Wallet::class);
}
```

nd in the wallets table, I can have multiple wallets assigned to the same user_id. I want to display a column in the user list where those wallets appear.

I’m trying with this, but I only see empty content:

```
        CRUD::addColumn([
            'name' => 'wallets',
            'type' => 'relationship',
            'label' => 'Wallets',
            'entity' => 'wallets',
            'attribute' => 'address',
            'model' => "App\Models\Wallet",
        ]);
```

I think I’m misunderstanding something about the relationships, in fact, the relations_count isn’t working well for me either.

Could it be that I’m missing something?

Thanks

**Comment by pxpm** [comment]
@infiniteroles does your `wallets` table have an `address` column ? 

If it's a simple text column, just do `CRUD::addColumn('wallets.address')`. 

Let me know if that helps 🙏

**Reply by infiniteroles** [reply]
Hello, first of all, thank you very much for your response. Unfortunately, it hasn’t worked. The column is still appearing empty. I have checked that the “address” field exists in the “wallets” table, and that the users I am testing with have wallets associated with their user_id.

This is my model for wallets table:
    public $table = "wallets";

```

    protected $fillable = [
        'user_id',
        'address',
        'name',
    ];


    public function user()
    {
        return $this->belongsTo(User::class);
    }
```
and I have this also in mu User model:

```
    public function wallets()
    {
        return $this->hasMany(Wallet::class);
    }
```

What I need is to display the list of wallets for each user in the user list in the UserCrudController.

Thank you very much again.

**Reply by pxpm** [reply]
No issues. Let's work out what's wrong here. 👍  

I've just tested it and it seems to be working, so let's try to find the culprit. This is my setup (you can find this setup in our demo) https://github.com/Laravel-Backpack/demo/blob/ace27507e84c3ca4c32f31a7648b104e73a69c4a/app/Models/Monster.php#L137
```php
// the relation in the model
public function postalboxer()
{
    return $this->hasMany(\App\Models\PostalBoxer::class);
}

// and the column in the controller
CRUD::column('postalboxer');
```

![image](https://github.com/user-attachments/assets/4382220d-9b63-41b7-8943-ea49c4ff4bfc)

If you go "edit" that entry you say it has the relationships set, does the related entries appear in the field ? 
![image](https://github.com/user-attachments/assets/1a26d832-6f92-4bf8-bffd-423eec6f70a5)


Cheers

**Reply by infiniteroles** [reply]
I get this error:

Call to undefined method App\Models\Wallet::identifiableAttribute()

and I have this in my user model:
```
    public function wallets()
    {
        return $this->hasMany(Wallet::class);
    }
```

and this in my crud file:

`CRUD::column('wallets');`

I also tested creating a new function with a different name, but I had the same problem.



Thanks so much

**Reply by pxpm** [reply]
From that error I think you forgot to add the `CrudTrait` in your model. Can you confirm ?

**Reply by infiniteroles** [reply]
It was missing in Wallet Model. Now I donñ´t have any error, but wallet colum is empty. :-(

**Reply by pxpm** [reply, marked as answer]
Did you define the column just as `CRUD::column('wallets')->attribute('address')` and let Backpack guess the other stuff? 

Can you go to `vendor/backpack/crud/src/resources/views/crud/columns/select_multiple.blade.php` and add a `dd($column['value'])` before the `@endphp` tag ? 

Cheers

**Reply by infiniteroles** [reply]
Finally I found an error in mi DB, Tahnk you very much for your help!!!!!   dd() helped so much to fin my error.

**Reply by karandatwani92** [reply]
Hey @infiniteroles 

If you like Backpack so far, please give it a star on the [Github repo](https://github.com/Laravel-Backpack/CRUD). Thanks
